src/components/RollCallList/RollCallChooseQuestion.vue

```diff
@@ -0,0 +1,100 @@
+<script setup lang="ts">
+import { ref, computed, onMounted, watch } from 'vue'
+import { apiClient } from '@/api/apiClient'
+import type { components } from '@/api/schema'
+type Question = components['schemas']['QuestionResponse']
+type QuestionGroup = components['schemas']['QuestionGroupResponse']
+type SelectQuestion =
+  | components['schemas']['SingleChoiceAnswerResponse']
+  | components['schemas']['MultipleChoiceAnswerResponse']
+
+const target = defineModel<string>({ required: true })
+interface itemType {
+  id: string
+  title: string
+  data: { questionId: number; optionId: number }
+}
+const { questionGroups } = defineProps<{
+  questionGroups: QuestionGroup[]
+}>()
+const filteredQuestionGroups = computed(() =>
+  questionGroups
+    .map((g) => ({
+      ...g,
+      questions: g.questions.filter((q: Question) => q.type === 'single' || q.type === 'multiple'),
+    }))
+    .filter((g) => g.questions.length > 0),
+)
+const selectedIDs = ref<itemType[]>([])
+const answers = ref<SelectQuestion[]>([])
+const fetchAnswers = async () => {
+  return (
+    await Promise.all(
+      filteredQuestionGroups.value
+        .map((g) => g.questions)
+        .map(async (qs) => {
+          return Promise.all(
+            qs.map(async (q) => {
+              const { data } = await apiClient.GET('/api/admin/questions/{questionId}/answers', {
+                params: { path: { questionId: q.id } },
+              })
+              return data as SelectQuestion[]
+            }),
+          )
+        }),
+    )
+  ).flat(2)
+}
```

Comment:
やや複雑なロジックになっているので、このようにしてはどうでしょうか
```suggestion
const fetchAnswersForQuestion = async (questionId: number) => {
  const { data } = await apiClient.GET('/api/admin/questions/{questionId}/answers', {
    params: { path: { questionId } },
  })
  return (data as SelectQuestion[]) ?? []
}

const fetchAnswers = async (questionIds: number[]) => {
  const responses = await Promise.all(questionIds.map((id) => fetchAnswersForQuestion(id)))
  return responses.flat()
}

onMounted(async () => {
  const questionIds = filteredQuestionGroups.value.flatMap((g) => g.questions.map((q) => q.id))
  answers.value = await fetchAnswers(questionIds)
})
```